client/store/allProducts.js

```diff
@@ -50,7 +50,7 @@ export default (state = initialState, action) => {
     case GET_PRODUCTS:
       return action.requestedProducts
     case DELETE_PRODUCT:
-      return {...state}
+      return [...state]
```

Comment:
When you dispatch removeProduct with the requestedProduct from the database, wouldn't you need to filter the current state to remove action.requestedProduct? Otherwise, isn't this just sending back the current state?